### ISSUE TITLE
Cart for each user

### ISSUE THREAD

**Issue by botoxparty** [seb86/cart-rest-api-for-woocommerce#39]
Hey,

So i've been wondering how to I use this plugin to work with carts from specific users? It's not clear in the documentation.

I would have assumed making a request such as..

`http://domain.com/wp-json/wc/v2/cart/ -u consumer_key:consumer_secret`

## Prerequisites

- [ X ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ X ] The issue still exists against the latest `master` branch of Cart REST-API for WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [ X ] I have attempted to find the simplest possible steps to reproduce the issue.
- [ ] I have included a failing test as a pull request (Optional)
- [ X ] I have installed the requirements to run this plugin.

## Steps to reproduce the issue

1. Install Cart REST API Plugin to fresh Woo and Wordpress instance.
2. Make a request to the Cart API e.g. http://domain.com/wp-json/wc/v2/cart/

## Expected/actual behaviour

When I follow those steps, I see...

I get a successful result such as "Cart is empty"

I was expecting to see...

No response as i'm not passing any consumer key or secret, or authentication in the request.

## Isolating the problem

- [ X ] This bug happens with only WooCommerce and Cart REST-API for WooCommerce plugin are active.
- [ ] This bug happens with a default WordPress theme active.
- [ ] This bug happens with the WordPress theme Storefront active.
- [ X ] This bug happens with the latest release of WooCommerce active.
- [ X ] I can reproduce this bug consistently using the steps above.

## WordPress Environment

<details>
```
Go to "WooCommerce > System Status then copy and paste the system status report here.
```
</details>

**Comment by seb86** [comment]
The documentation is outdated and I will get round to updating it when I can afford to spend time on it.

The API works with or without authentication but works best when not authenticating as a user. The API shows what you have in the cart, either added directly from your site or via the API.

The API acts just like the cart on the site. No more, no less.

**Comment by botoxparty** [comment]
So this command `http://domain.com/wp-json/wc/v2/cart/ -u consumer_key:consumer_secret` would pull the cart for the user that i'm sending through?

**Comment by seb86** [comment]
Correct

**Comment by botoxparty** [comment]
Great, thanks seb. I'll try that out tomorrow. 

Also happy to contribute to the docs, I'll let you know how it all goes.

**Comment by JosephCarrington** [comment]
So in order to  have an individual subscriber of our site have their cart accessed via the API, we would need to automatically generate each user a consumer_key and consumer_secret? Or have them authorize our app like this? http://woocommerce.github.io/woocommerce-rest-api-docs/images/woocommerce-auth-endpoint-flow.png

**Comment by botoxparty** [comment]
@JosephCarrington yep exactly, i've written a function that generates the keys whenever a user registers in WordPress and stores them in their user info, and they are then passed in a JWT token to the client.

**Comment by BlueCloud02** [comment]
I have just installed the plugin. But without the documentation, I have some problems.
I've tested the API with Postman and all works fine. But when I use it inside my website (without authentification), it is not working. (I am not creating my own web app but I want to use the API inside the initial wordpress-woocommerce website).
Do I have to use authentification? @botoxparty  could you precise how you did that?

**Comment by seb86** [comment]
@BlueCloud02 Documentation has moved. You can access it here: https://co-cart.github.io/co-cart-docs/

**Comment by seb86** [comment]
The WooCommerce cart does use cookies and sessions. It's just encrypted as a hash, I think. You will have to use the cart normally and then look for what the value of the cookie WooCommerce saves to confirm.

**Comment by botoxparty** [comment]
@seb86 Hmm i understand that the cart uses cookies and sessions however getting the API to refer to these particular sessions or cookies isn't documented. I'd be happy to help write the docs to perform these actions if you're willing to chat about it over slack or something?

This project has helped me a lot with my work and am very happy to contribute to driving it forward.

**Comment by seb86** [comment]
@botoxparty Sorry forgot to provide a link to the documentation. https://github.com/co-cart/co-cart-docs which I have just updated. More is to come but you are welcome to contribute to it now.
